### PR TITLE
Adding `--with-raw` flag to convert commands for BurpSuite, JFrog Xray, Nessus, Nikto, Sarif, ScoutSuite, Twistlock, Zap

### PR DESCRIPTION
**Pull request by charleshu-8**
Signed-off-by: charleshu-mitre <chu@mitre.org>

### CODE REVIEW

**Comment by aaronlippold** [comment]
i would prefer we use a more standard naming convention like `--with-raw`

**Comment by aaronlippold** [comment]
or perhaps more simply `--raw` but perhaps `--with-raw` is better.

File: README.md [hunk 1]
```diff
@@ -16,42 +16,71 @@ The SAF CLI is the successor to [Heimdall Tools](https://github.com/mitre/heimda
...
+      - [Thresholds](#thresholds)
+    - [Generate](#generate)
+      - [CKL Templates](#ckl-templates)
+      - [InSpec Metadata](#inspec-metadata)
+      - [Thresholds](#thresholds-1)
+      - [Spreadsheet (csv/xlsx) to InSpec](#spreadsheet-csvxlsx-to-inspec)
+      - [XCCDF to InSpec Stub](#xccdf-to-inspec-stub)
+      - [Other](#other)
+        - [Notes](#notes)
+        - [Mapping Files](#mapping-files)
```
Comment (camdenmoors) [inline README.md:78]: hmm
There seems to be some markdown tags being shown inside of the rendered text. Also please confirm with Eugene if this layout is ok if you haven't

Comment (charleshu-8) [reply README.md]: Eugene made some changes and gave his approval, we should be good on the ToC changes.

File: src/commands/convert/sarif2hdf.ts [hunk 2]
```diff
@@ -4,16 +4,17 @@ import {SarifMapper as Mapper} from '@mitre/hdf-converters'
...
 
-  static examples = ['saf convert sarif2hdf -i sarif-results.json -o output-hdf-name.json']
+  static examples = ['saf convert sarif2hdf -i sarif-results.json -o output-hdf-name.json -w']
 
   static flags = {
     help: Flags.help({char: 'h'}),
-    input: Flags.string({char: 'i', required: true}),
-    output: Flags.string({char: 'o', required: true}),
+    input: Flags.string({char: 'i', required: true, description: 'Input SARIF file'}),
+    output: Flags.string({char: 'o', required: true, description: 'Output HDF file'}),
```
Comment (Rlin232) [inline src/commands/convert/sarif2hdf.ts:16]: These descriptions are not consistent with the README (but https://github.com/mitre/saf/pull/526 solves this)

**Review by Rlin232 — COMMENTED** [review]
Changes and additions all look good to me. Only concern I have is if https://github.com/mitre/saf/pull/526 gets merged in and some of this ends up being redundant/conflicting.

File: README.md [hunk 3]
```diff
@@ -346,10 +371,10 @@ convert burpsuite2hdf       Translate a BurpSuite Pro XML file into a Heimdall
   OPTIONS
     -i, --input=xml            Input BurpSuite Pro XML File
     -o, --output=output        Output HDF JSON File
-
+    -w, --with-raw             Append raw input file to HDF output file
```
Comment (Amndeep7) [inline README.md:374]: `Include raw input file in HDF JSON file`

File: README.md [hunk 4]
```diff
@@ -407,12 +432,13 @@ convert fortify2hdf         Translate a Fortify results FVDL file into a Heimdal
...
-  -i, --input=<value>   (required) Input Twistlock file
-  -o, --output=<value>  (required) Output HDF file
+    -h, --help            Show CLI help.
+    -i, --input=<value>   (required) Input Twistlock file
+    -o, --output=<value>  (required) Output HDF file
+    -w, --with-raw        Append raw input file to HDF output file
 
   EXAMPLES
-    saf convert twistlock2hdf -i twistlock.json -o output-hdf-name.json
+    saf convert twistlock2hdf -i twistlock.json -o output-hdf-name.json -w
```
Comment (Amndeep7) [inline README.md:441]: We want that to be there, but we don't want to promote it as default behavior.  Consequently, don't include `-w` in the example commands.

File: src/commands/convert/burpsuite2hdf.ts [hunk 5]
```diff
@@ -4,16 +4,17 @@ import {BurpSuiteMapper as Mapper} from '@mitre/hdf-converters'
 import {checkInput, checkSuffix} from '../../utils/global'
 
 export default class Burpsuite2HDF extends Command {
-  static usage = 'convert burpsuite2hdf -i, --input=XML -o, --output=OUTPUT'
+  static usage = 'convert burpsuite2hdf -i, --input=XML -o, --output=OUTPUT -w, --with-raw'
```
Comment (Amndeep7) [inline src/commands/convert/burpsuite2hdf.ts:7]: Like @Rlin232 mentioned, don't worry about the usage statements for now.  They're going to be drastically revised in an upcoming PR.

File: test/commands/convert/zap2hdf.test.ts [hunk 6]
```diff
@@ -24,3 +24,24 @@ describe('Test zap', () => {
...
+  .stdout()
+  .command(['convert zap2hdf', '-i', path.resolve('./test/sample_data/zap/sample_input_report/webgoat.json'), '-n', 'http://mymac.com:8191', '-o', `${tmpobj.name}/zaptest-webgoat.json`, '-w'])
+  .it('hdf-converter withraw output test - webgoat', () => {
+    const test = JSON.parse(fs.readFileSync(`${tmpobj.name}/zaptest-webgoat.json`, 'utf8'))
+    const sample = JSON.parse(fs.readFileSync(path.resolve('./test/sample_data/zap/zap-webgoat-hdf-withraw.json'), 'utf8'))
+    expect(omitHDFChangingFields(test)).to.eql(omitHDFChangingFields(sample))
+  })
+  test
+  .stdout()
+  .command(['convert zap2hdf', '-i', path.resolve('./test/sample_data/zap/sample_input_report/zero.webappsecurity.json'), '-n', 'http://zero.webappsecurity.com', '-o', `${tmpobj.name}/zaptest-webappsecurity.json`, '-w'])
```
[the remark this answers is not in the document]
Comment (charleshu-8) [reply test/commands/convert/zap2hdf.test.ts:41]: It passes the GitHub Action lint test, but I can prettify it if need be

File: src/commands/convert/burpsuite2hdf.ts [hunk 7]
```diff
@@ -4,16 +4,17 @@ import {BurpSuiteMapper as Mapper} from '@mitre/hdf-converters'
...
   static description = 'Translate a BurpSuite Pro XML file into a Heimdall Data Format JSON file'
 
-  static examples = ['saf convert burpsuite2hdf -i burpsuite_results.xml -o output-hdf-name.json']
+  static examples = ['saf convert burpsuite2hdf -i burpsuite_results.xml -o output-hdf-name.json -w']
 
   static flags = {
     help: Flags.help({char: 'h'}),
     input: Flags.string({char: 'i', required: true}),
     output: Flags.string({char: 'o', required: true}),
+    'with-raw': Flags.boolean({char: 'w', required: false}),
```
[the remark this answers is not in the document]
Comment (charleshu-8) [reply src/commands/convert/burpsuite2hdf.ts:17]: Just wanted to clarify, is snake case or kebab case the correct scheme to use here?

File: README.md [hunk 8]
```diff
@@ -16,42 +16,67 @@ The SAF CLI is the successor to [Heimdall Tools](https://github.com/mitre/heimda
...
-        *  [HDF to Splunk](#hdf-to-splunk)
-        *  [HDF to XCCDF](#hdf-to-xccdf)
-        *  [HDF to CSV](#hdf-to-csv)
-        *  [HDF to DISA Checklist](#hdf-to-checklist)
-  * [View](#view) - Identify overall security status and deep-dive to solve specific security defects
-  * [Validate](#validate) - Verify pipeline thresholds
-  * [Generate](#generate) - Generate InSpec validation code, set pipeline thresholds, and generate options to support other saf commands.
-  * Scan - Visit https://saf.mitre.org/#/validate to explore and run inspec profiles
-  * Harden - Visit https://saf.mitre.org/#/harden to explore and run hardening scripts
+- [Security Automation Framework CLI](#security-automation-framework-cli)
```
[the remark this answers is not in the document]
Comment (charleshu-8) [reply README.md:19]: I had ran this change through with @ejaronne and he had approved of it, but if need be I can revert it. However, if possible, could I maintain the links to some of the subsections under the item in the ToC for ease of access and use?

For examples, the current ToC is structured as such:

* [View](#view) - Identify overall security status and deep-dive to solve specific security defects
* [Validate](#validate) - Verify pipeline thresholds
...
* Scan - Visit https://saf.mitre.org/#/validate to explore and run inspec profiles
* Harden - Visit https://saf.mitre.org/#/harden to explore and run hardening scripts

Could I hybridize my changes with the current format to produce something like this:

* [View](#view) - Identify overall security status and deep-dive to solve specific security defects
  * [Heimdall](#heimdall)
  * [Summary](#summary)
* [Validate](#validate) - Verify pipeline thresholds
  * [Thresholds](#thresholds)
...
* Scan - Visit https://saf.mitre.org/#/validate to explore and run inspec profiles
* Harden - Visit https://saf.mitre.org/#/harden to explore and run hardening scripts

[the remark this answers is not in the document]
Comment (Amndeep7) [reply README.md:19]: Let's discuss on monday what this hybrid form will look like.

File: src/commands/convert/burpsuite2hdf.ts (hunk 7 above)
[the remark this answers is not in the document]
Comment (Amndeep7) [reply src/commands/convert/burpsuite2hdf.ts:17]: i'm dumb, i meant kebab case

File: test/commands/convert/zap2hdf.test.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (Amndeep7) [reply test/commands/convert/zap2hdf.test.ts:41]: please do so

**Comment by charleshu-8** [comment]
HANDOFF: Building error (visible in GitHub Action); need to fix before merging

**Comment by charleshu-8** [comment]
Unifying all withraw additions into single PR

**Pull request closed without merging**